clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -327,25 +345,22 @@ private MetadataCache handleMetadataResponse(MetadataResponse metadataResponse,
     private void updatePartitionInfo(String topic,
                                      MetadataResponse.PartitionMetadata partitionMetadata,
                                      boolean hasReliableLeaderEpoch,
-                                     Consumer<MetadataCache.PartitionInfoAndEpoch> partitionInfoConsumer) {
+                                     Consumer<MetadataResponse.PartitionMetadata> partitionInfoConsumer) {
```

Comment:
Given the updated code, do we actually need this consumer? Or could this method be changed to return `Optional<PartitionInfo>`/`PartitionInfo`?

Reply:
Good call. That does seem a bit simpler.